pwiz_tools/Shared/Common/Collections/Range.cs

```diff
@@ -29,4 +32,32 @@ public Range(int start, int end) : this()
         public int End { get; private set; }
         public int Length { get { return End - Start; } }
     }
+
+    public struct Interval<T> where T:IComparable

```

Comment:
Given that one of the fields in this struct is a Delegate, it kind of defeats the (memory overhead saving) purpose of it being a struct.

Reply:
Well, I think struct copy constructor performs a shallow copy, so a new heap allocation happens only during initial construction, but not at assignment or parameter passing.